lib/active_record/virtual_attributes.rb

```diff
@@ -77,6 +77,10 @@ def virtual_attribute(name, type, through: nil, uses: nil, arel: nil, source: na
           #   Because we may not have loaded the class yet
           #   And we definitely have not loaded the database yet
           arel ||= virtual_delegate_arel(source, to_ref)
+        elsif block_given?
+          define_method(name) do
+            has_attribute?(name) ? self[name] : instance_eval(&block)
```

Comment:
this is so nice.

We can also auto add a group or anything else that may change over time.
The current implementation doesn't allow us to do any common manipulation of the block return values